keys.go

```diff
@@ -133,22 +133,116 @@ type KeyVersion struct {
 	CreationDate *time.Time `json:"creationDate,omitempty"`
 }
 
+// This function returns a string so we can pass extra info not in the key struct if needed
+type CreateKeyOption func(k *Key) string
+
+func WithExpiration(expiration *time.Time) CreateKeyOption {
+	return func(key *Key) string {
+		key.Expiration = expiration
+		return "expiration"
+	}
+}
+
+func WithDescription(description string) CreateKeyOption {
+	return func(key *Key) string {
+		key.Description = description
+		return "description"
+	}
+}
+
+func WithPayload(payload, encryptedNonce, iv string, sha1 bool) CreateKeyOption {
+	return func(key *Key) string {
+		ret := "payload"
+		key.Payload = payload
+		if !key.Extractable {
+			algorithm := AlgorithmRSAOAEP256
+			if sha1 {
+				algorithm = AlgorithmRSAOAEP1
+			}
+			key.EncryptedNonce = encryptedNonce
+			key.IV = iv
+			key.EncryptionAlgorithm = algorithm
+			ret += "-" + algorithm
+		}
+		return ret
+	}
+}
+
+func WithAliases(aliases []string) CreateKeyOption {
+	return func(key *Key) string {
+		key.Aliases = aliases
+		return "aliases"
+	}
+}
+
+func WithPolicy(policy *Policy) CreateKeyOption {
+	return func(key *Key) string {
+		/*
+			Setting the value of rotationInterval to -1 in case user passes 0 value
+			as we want to retain the param `interval_month` after marshalling
+			so that we can get correct error msg from REST API saying interval_month should be between 1 to 12
+			Otherwise the param would not be sent to REST API in case of value 0
+			and it would throw error saying interval_month is missing
+		*/
+		if policy.Rotation != nil && policy.Rotation.Interval == 0 {
+			policy.Rotation.Interval = -1
+		}
+		key.Rotation = policy.Rotation
+		key.DualAuthDelete = policy.DualAuth
+		return "policy"
+	}
+}
+
+func WithTags(tags []string) CreateKeyOption {
+	return func(key *Key) string {
+		key.Tags = tags
+		return "tags"
+	}
+}
+
+func (c *Client) CreateKeyWithOptions(ctx context.Context, name string, extractable bool, options ...CreateKeyOption) (*Key, error) {
+	key := &Key{
+		Name:        name,
+		Type:        keyType,
+		Extractable: extractable,
+	}
+	overrides := false
+	for _, opt := range options {
+		if opt != nil {
+			optName := opt(key)
+			if optName == "policy" {
+				overrides = true
+			}
+		}
+	}
+
+	path := keysPath
+	if overrides {
+		path = keysWithPolicyOverridesPath
+	}
+	return c.createKeyResource(ctx, *key, path)
+}
+
 // CreateKey creates a new KP key.
 func (c *Client) CreateKey(ctx context.Context, name string, expiration *time.Time, extractable bool) (*Key, error) {
-	return c.CreateImportedKey(ctx, name, expiration, "", "", "", extractable)
+	return c.CreateKeyWithOptions(ctx, name, extractable, WithExpiration(expiration))
```

Comment:
It looks like an empty string payload is just omitted by the json marshalling, but that behaviour seems unintuitive, so i made it explicit so that payload is nil.